url47ebb27197_asirbu_CorrectiveEvolution_tgz-pJ8-eu_fbk_soa_evolution_engine_impl_test_ProblemToSMVFuncTestJ8/src/eu/fbk/soa/evolution/CorrectiveEvolution.java

```diff
@@ -0,0 +1,76 @@
+package eu.fbk.soa.evolution;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.log4j.Logger;
+
+import eu.fbk.soa.evolution.engine.CorrectiveEvolutionEngine;
+import eu.fbk.soa.evolution.engine.EngineFactory;
+import eu.fbk.soa.process.Goal;
+import eu.fbk.soa.process.ProcessModel;
+
+
+public class CorrectiveEvolution {
+	
+	private static Logger logger = Logger.getLogger(CorrectiveEvolution.class);
+	
+	private String path = "";
+	
+	private boolean experimentMode = false;
+		
+	public CorrectiveEvolution(String path) {
+		this.setOutputPath(path);		
+	}
+	
+	public void setOutputPath(String path) {
+		this.path = path;
+		File outDir = new File(path);
+		if (!outDir.exists()) {
+			outDir.mkdirs();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Confirm that `mkdirs` was successful. Handle any failures that happened. If you fail silently, then you might encounter errors that are harder to debug. See the following diffs on GitHub that make this code change: [diff1](https://github.com/qunarcorp/qmq/commit/12f1d9d6a57a02bc2415cb2af9f4388a77b12c02#diff-6b67be6e17f24fdeddb7c725ac8ec5600d6e63be0bdca0963a83d530e6d55e35L36), [diff2](https://github.com/apache/jmeter/commit/1ec61206779b3f84df18516682b4791bf015178f#diff-cd74304763c097fbc085ac2bc952b46ab3843ac825cacacad89e8064296e955aL330)

Learn more: [File.mkdirs in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdirs--)